packages/flutter_localizations/test/material/translations_test.dart

```diff
@@ -174,8 +174,8 @@ void main() {
     expect(localizations.selectedRowCountTitle(0), 'បាន​ជ្រើស​រើស​ធាតុ 0');
     expect(localizations.selectedRowCountTitle(1), 'បាន​ជ្រើស​រើស​ធាតុ 1');
     expect(localizations.selectedRowCountTitle(2), 'បាន​ជ្រើស​រើស​ធាតុ 2');
-    expect(localizations.selectedRowCountTitle(10000), 'បាន​ជ្រើស​រើស​ធាតុ 10.000');
-    expect(localizations.selectedRowCountTitle(123456789), 'បាន​ជ្រើស​រើស​ធាតុ 123.456.789');
+    expect(localizations.selectedRowCountTitle(10000), 'បាន​ជ្រើស​រើស​ធាតុ 10,000');
+    expect(localizations.selectedRowCountTitle(123456789), 'បាន​ជ្រើស​រើស​ធាតុ 123,456,789');
```

Comment:
@goderbauer and I did some research:

1. Wikipedia article indicates Khmer uses commas as the thousands separator: https://en.wikipedia.org/wiki/Khmer_numerals#Numbers_from_100_to_10,000,000
2. Google Translate uses commas as the thousands separator: https://translate.google.com/?sl=en&tl=km&text=I%20have%201%2C000%20apples&op=translate
3. Gemini agrees it should be a comma: `The Khmer language uses a comma (,) as its thousands separator. However, you might also see spaces used as a thousands separator, especially in less formal contexts.`

Reply:
(Drive-by reviewing, feel free to ignore) ICU says: 

https://www.localeplanet.com/icu/km-KH/index.html

> Numbers
> * Grouping separator: `.`
> * Decimal separator: `,`

Sauce:

https://github.com/unicode-org/icu/blob/b396885aae1f515df35688ea72943583c0268836/icu4c/source/data/locales/km.txt#L17-L22

Reply:
Sometimes software is too hard

Reply:
Looks like this change is correct, per: https://github.com/flutter/flutter/pull/145167#issuecomment-1998170061

Reply:
Hah, and people complain about timezones :P What an amazing "Winning" column in that link :P

Reply:
This is where the separators flipped:

https://unicode.org/cldr/charts/44/delta/km.html

(Looks like ICU4c hasn't regenerated data from the latest CLDR yet?)